Ensure that the cell originating in spanning column respect min/max-width of 1px

Ahmad-S792 opened the pull request:
#### 9c19b008c6622e71dfe649ff8a75a267a99125b7
<pre>
Ensure that the cell originating in spanning column respect min/max-width of 1px
<a href="https://bugs.webkit.org/show_bug.cgi?id=291662">https://bugs.webkit.org/show_bug.cgi?id=291662</a>

Reviewed by Antti Koivisto and Alan Baradlay.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Partial Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/4661416ff22985013680db8901b01ca79c026c87">https://chromium.googlesource.com/chromium/src.git/+/4661416ff22985013680db8901b01ca79c026c87</a>

This patch covers a case to ensure that the cell originating in the spanning
column have a min/max width of at least 1px for it.

* Source/WebCore/rendering/AutoTableLayout.cpp:
(WebCore::AutoTableLayout::recalcColumn):

&gt; Progressions:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-003-expected.txt:

&gt; Rebaselines:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug1302-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1302-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1302-expected.txt:

Canonical link: <a href="https://commits.webkit.org/293812@main">https://commits.webkit.org/293812@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7809a1331f45d4d254551afa96e467f5ee5210f9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19641 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9933 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105121 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50574 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102034 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19947 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/28096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/76122 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/33206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103000 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/15238 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90311 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56481 "Found 1 new API test failure: /WPE/TestWebKitWebView:/webkit/WebKitWebView/is-playing-audio (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15043 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8301 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49943 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/84969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8386 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107481 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27106 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19818 "Unable to confirm if test failures are introduced by change, retrying build") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/85077 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27469 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86511 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/84601 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21488 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29285 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/7011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/20942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27043 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32271 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26854 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30170 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28413 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/AutoTableLayout.cpp:
```diff
@@ -124,6 +124,9 @@ void AutoTableLayout::recalcColumn(unsigned effCol)
                 } else if (!effCol || section->primaryCellAt(i, effCol - 1) != cell) {
+                    // If a cell originates in this spanning column ensure we have a min/max width of at least 1px for it.
+                    columnLayout.minLogicalWidth = std::max<float>(columnLayout.minLogicalWidth, cell->maxPreferredLogicalWidth() ? 1 : 0.f);
+
```
why not 1.f if we already return 0.f? -and drop <float>
`columnLayout.minLogicalWidth = std::max(columnLayout.minLogicalWidth, cell->maxPreferredLogicalWidth() ? 1.f : 0.f);
`

Ahmad-S792 replied:
@alanbaradlay - Done. :-)

Commit:
Ensure that the cell originating in spanning column respect min/max-width of 1px
https://bugs.webkit.org/show_bug.cgi?id=291662

Reviewed by Antti Koivisto and Alan Baradlay.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Partial Merge: https://chromium.googlesource.com/chromium/src.git/+/4661416ff22985013680db8901b01ca79c026c87

This patch covers a case to ensure that the cell originating in the spanning
column have a min/max width of at least 1px for it.

* Source/WebCore/rendering/AutoTableLayout.cpp:
(WebCore::AutoTableLayout::recalcColumn):

> Progressions:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-003-expected.txt:

> Rebaselines:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/glib/tables/mozilla/bugs/bug1302-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1302-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug10633-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1302-expected.txt:

Canonical link: https://commits.webkit.org/293812@main
